tests/unit/test_cv_device_tools.py

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/python
+# coding: utf-8 -*-
+# pylint: disable=logging-format-interpolation
+# pylint: disable=dangerous-default-value
+# pylint:disable=duplicate-code
+# flake8: noqa: W503
+# flake8: noqa: W1202
+# flake8: noqa: R0801
+
+from unittest.mock import call
+import pytest
+from tests.data.device_tools_unit import device_data, device_data_invalid
+from ansible_collections.arista.cvp.plugins.module_utils.device_tools import DeviceInventory, CvDeviceTools
+
+
+# status = ''
+
+@pytest.fixture
+def setup(apply_mock, mock_cvprac):
+    """
+    setup - setup method to apply mocks and patches
+    """
+    mock_ansible_module, mock__get_device = apply_mock(TestState.MOCK_LIST)
+    dummy_cvprac, mock_cvpClient = mock_cvprac
+    mock_cvpClient.api.device_decommissioning.side_effect = dummy_cvprac.device_decommissioning
+    mock_cvpClient.api.device_decommissioning_status_get_one.side_effect = dummy_cvprac.device_decommissioning_status_get_one
+    mock_cvpClient.api.reset_device.side_effect = dummy_cvprac.reset_device
+    mock_cvpClient.api.delete_device.side_effect = dummy_cvprac.delete_device
+    cv_tools = CvDeviceTools(mock_cvpClient, mock_ansible_module)
+
+    return mock_ansible_module, mock__get_device, cv_tools
+
+
+@pytest.mark.state
+class TestState():
```

Comment:
Could you separate `test_state_absent()`, `test_state_factory_reset()` and `test_state_provisioning_reset()` into it's own classes and add more tests?

Reply:
Sure, I will update.